packages/lexicon-resolver/README.md

```diff
@@ -0,0 +1,79 @@
+# @atproto/lexicon-resolver
+
+ATProto Lexicon resolution
+
+[![NPM](https://img.shields.io/npm/v/@atproto/lexicon-resolver)](https://www.npmjs.com/package/@atproto/lexicon-resolver)
+[![Github CI Status](https://github.com/bluesky-social/atproto/actions/workflows/repo.yaml/badge.svg)](https://github.com/bluesky-social/atproto/actions/workflows/repo.yaml)
+
+## Usage
+
+This package may be used to determine the DID authority for a Lexicon based on its NSID, and to resolve a Lexicon from its NSID based on [Lexicon Resolution](https://atproto.com/specs/lexicon#lexicon-publication-and-resolution) from the network. Resolutions always verify the inclusion proof for the Lexicon schema document published to the ATProto network.
+
+```ts
+import {
+  resolveLexicon,
+  resolveLexiconDidAuthority,
+} from '@atproto/lexicon-resolution'
+
+// Which DID is the authority over this Lexicon?
+const didAuthority = await resolveLexiconDidAuthority('app.bsky.feed.post')
+// Resolve the Lexicon document with resolution details
+const resolved = await resolveLexicon('app.bsky.feed.post')
+/**
+ * {
+ *   commit: {
+ *     did: 'did:plc:4v4y5r3lwsbtmsxhile2ljac',
+ *     rev: '3lnlpukgipj2c',
+ *     sig: Uint8Array(64),
+ *     ...
+ *   },
+ *   uri: AtUri(at://did:plc:4v4y5r3lwsbtmsxhile2ljac/com.atproto.lexicon.schema/app.bsky.feed.post),
+ *   cid: CID(bafyreidgbehqwweghrrddfu6jgj7lyr6fwhzgazhirnszdb5lvr7iynkiy),
+ *   nsid: NSID('app.bsky.feed.post'),
+ *   lexicon: {
+ *     '$type': 'com.atproto.lexicon.schema',
+ *     lexicon: 1
+ *     id: 'app.bsky.feed.post',
+ *     defs: { main: [Object], ... },
+ *   }
+ * }
+ */
+```
+
+### With identity caching
+
+Identity data is used in order to fetch and verify record contents. The @atproto/identity package can be used to offer more control over caching and other behaviors of identity lookups.
+
+```ts
+import { IdResolver, MemoryCache } from '@atproto/identity'
+import { resolveLexicon } from '@atproto/lexicon-resolution'
+
+const idResolver = new IdResolver({
+  didCache: new MemoryCache(),
+})
+
+const resolved = await resolveLexicon('app.bsky.feed.post', {
+  idResolver,
+})
+```
+
+### With DID authority override
+
+You may specify a specific DID authority you'd like to use to perform a Lexicon resolution, overriding ATProto's DNS-based authority over Lexicons. This is described in some more detail in [Authority and Control](https://atproto.com/specs/lexicon#authority-and-control).
+
+```ts
+import { resolveLexicon } from '@atproto/lexicon-resolution'
+
+const resolved = await resolveLexicon('app.bsky.feed.post', {
+  didAuthority: 'did:plc:...',
```

Comment:
if this is the "resolution override" mechanism, should this be an array of DIDs, not just one?

Reply:
Actually this isn't quite the override mechanism, currently it's just a way to specify the authority you want to pull the lexicon from.  I think it's nice to have, and we'll also use it for the recursive lexicon resolution to use cached authority lookups.